src/main/java/com/teradata/tpcds/ScalingInfo.java

```diff
@@ -74,16 +74,24 @@ private long computeCountUsingStaticScale()
         return getRowCountForScale(1);
     }
 
-    private long computeCountUsingLogScale(int scale)
+    private long computeCountUsingLogScale(double scale)
     {
         int scaleSlot = getScaleSlot(scale);
         long delta = getRowCountForScale(DEFINED_SCALES[scaleSlot]) - getRowCountForScale(DEFINED_SCALES[scaleSlot - 1]);
-        float floatOffset = (float) (scale - DEFINED_SCALES[scaleSlot - 1]) / (float) (DEFINED_SCALES[scaleSlot] - DEFINED_SCALES[scaleSlot - 1]);
+        double floatOffset = (scale - DEFINED_SCALES[scaleSlot - 1]) / (DEFINED_SCALES[scaleSlot] - DEFINED_SCALES[scaleSlot - 1]);
 
-        return (int) (floatOffset * (float) delta) + getRowCountForScale(DEFINED_SCALES[0]);
+        long baseRowCount = 0;
+        if (scale < 1.0) {
+            baseRowCount = getRowCountForScale(DEFINED_SCALES[0]);
+        }
+        else {
+            baseRowCount = getRowCountForScale(DEFINED_SCALES[1]);
+        }
+        long count = (long) (floatOffset * (float) delta) + baseRowCount;
```

Comment:
I see, you're right. I'm going to revert to an int cast. In fact, I broke the unit tests on this line by shifting around the order of casting. Took me hours to figure it out. I've learned my lesson now and will will stick as closely to the spec code as possible ;)